subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -3810,16 +3806,12 @@ static void lwm2m_engine_service(struct k_work *work)
 		/* service is due */
 		if (timestamp > service_due_timestamp) {
```

Comment:
If timestamp == service_due_timestamp, the work will not be handled. engine_next_service_timeout_ms will return 0 and lwm2m_engine_service will be called repeatedly without the work being handled.
```suggestion
		if (timestamp >= service_due_timestamp) {
```

Reply:
@pieterjanc I will add this as a second patch.  Thank you for pointing out the zero case.